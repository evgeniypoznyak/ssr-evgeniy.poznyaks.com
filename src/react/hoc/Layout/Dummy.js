import Button from '@material-ui/core/Button';
import React from 'react';

export const Dummy = props => {
    const dummyClick = () => {
        console.log('Clicked!');
    };

    return (<div >
        I'm dummy div
        <Button href={null} onClick={dummyClick}>I'm button</Button>
    </div>);
};