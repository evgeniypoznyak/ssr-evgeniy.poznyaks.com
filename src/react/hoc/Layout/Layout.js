import React, {Fragment} from 'react';
// import Navbar from '../../Navbar';
import Header from '../../components/Layouts/Header/Header';
import Main from '../../components/Layouts/Main/Main';
import Footer from '../../components/Layouts/Footer/Footer';

const Layout = (props) => {
    return (
        <Fragment>
            <Header/>
            <Main>
                {props.children}
            </Main>
            <Footer/>
        </Fragment>
    );
};
export default Layout;
