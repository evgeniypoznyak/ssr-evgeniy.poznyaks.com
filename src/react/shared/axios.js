import axios from 'axios';


axios.interceptors.request.use(
    config => {
        let token = '';
        // eslint-disable-next-line no-undef
        if (__isBrowser__) {
            if (localStorage) {
                token = localStorage.getItem('token');
            }
        }

        config.headers['x-auth-token'] = token;
        return config;
    },
    error => Promise.reject(error),
);

export default axios;
