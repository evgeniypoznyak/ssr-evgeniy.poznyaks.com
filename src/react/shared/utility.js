export const updateObject = (oldObject, updatedProperties) => {
    return {
        ...oldObject,
        ...updatedProperties,
    };
};

export const checkValidity = (value, rules) => {
    let isValid = true;
    if (!rules) {
        return true;
    }
    if (typeof (value) == 'undefined' || value == null) {
        return false;
    }

    if (rules.required) {
        isValid = ('' + value).trim() !== '' && isValid;
    }

    if (rules.minLength) {
        isValid = ('' + value).trim().length >= rules.minLength && isValid;
    }

    if (rules.maxLength) {
        const lenght = ('' + value).trim().length;
        isValid = lenght <= rules.maxLength && lenght > 0 && value !== '' && isValid;
    }

    if (rules.isEmail) {
        // eslint-disable-next-line max-len
        const pattern = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
        isValid = pattern.test(value) && isValid;
    }

    if (rules.isNumeric) {
        const pattern = /^\d+$/;
        isValid = pattern.test(value) && isValid;
    }

    return isValid;
};

export const browserLanguage = function browserLocale() {
    if (navigator.languages && navigator.languages.length) {
        return navigator.languages[0];
    } else if (navigator.userLanguage) {
        return navigator.userLanguage;
    }
    return navigator.language;
};

export const stripHtmlTags = element => element.replace(/<\/?[^>]+(>|$)/g, '');
