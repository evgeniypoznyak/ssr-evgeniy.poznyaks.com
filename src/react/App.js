import React, {useEffect} from 'react';
import routes from './routes';
import {Route, Switch} from 'react-router-dom';
import NoMatch from './components/404/NoMatch';
import Layout from './hoc/Layout/Layout';

const App = (props) => {
    useEffect(() => {

    }, []);

    // Where is a Switch and is then StaticContent from router defined
    const reactRoutes = <Switch>
        {routes.map(({path, exact, component: Component, ...rest}) => (
            <Route key={path} path={path} exact={exact} render={(props) => (
                <Component {...props} {...rest} />
            )}/>
        ))}
        <Route render={(props) => <NoMatch {...props} />}/>
    </Switch>;

    return (
        <Layout>
            {reactRoutes}
        </Layout>
    );
};
export default App;
