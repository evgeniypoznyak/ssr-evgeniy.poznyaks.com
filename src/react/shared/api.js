import axios from './axios';
import {skills} from './initialData/skills';

const url = process.env.API_GATEWAY ||
    'https://api.evgeniy.poznyaks.com'
    // 'http://localhost:2222'
;
const apiPublic = process.env.API_GATEWAY_PUBLIC ||
    'https://api.evgeniy.poznyaks.com'
    // 'http://localhost:2222'
;

const selfUrl = process.env.SELF_URL ||
    ''
    // 'http://localhost:3001'
;

export async function fetchSkills(path = 'all') {
    const biography = await getBiography();
    try {
        console.log('[GET] processing url: ', `${url}/api/skills/`);
        const result = await axios.get(`${url}/api/skills/`);
        return {
            skills: result.data.skills,
            biography,
        };
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        console.log('returning regular skills...');
        return {skills, biography};
    }
}

export async function sendEmail(data) {
    try {
        const url = `${apiPublic}/api/email`;
        console.log('[POST] processing url: ', `${url}`);
        const result = await axios.post(url, data);
        return result.data;
    } catch (e) {
        const error = {
            status: e.response.data.status,
            message: e.response.data.message,
            errorMessage: e.message,
        };
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return error;
    }
}

export async function fetchBiography(path) {
    console.log(path);
    const url = '/biography';
    try {
        console.log('[GET] processing url: ', `${url}/`);
        const result = await axios.get(`${url}/`);
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        console.log('returning regular skills...');
        return {skills};
    }
}

export async function signInIntoApiGateway(data) {
    try {
        console.log('[POST] processing url: ', `${apiPublic}/api/auth`);
        const result = await axios.post(`${apiPublic}/api/auth`, {...data});
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return {token: ''};
    }
}

export async function verifyToken() {
    try {
        // eslint-disable-next-line no-undef
        if (__isBrowser__ && localStorage.getItem('token') && localStorage.getItem('token').length > 0) {
            return 'OK';
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return {token: ''};
    }
}

export async function verifyTokenApi() {
    try {
        const url = `${apiPublic}/api/auth/verify`;
        console.log('[POST] processing url: ', `${url}`);
        const result = await axios.post(url);
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return {token: ''};
    }
}

export async function patchSkill(skill) {
    try {
        const url = `${apiPublic}/api/skills`;
        console.log('[PATCH] processing url: ', `${url}`);
        const result = await axios.patch(url, {skill});
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return 'Skill not been patched';
    }
}

export async function deleteSkill(skillId) {
    try {
        const url = `${apiPublic}/api/skills/` + skillId;
        console.log('[DELETE] processing url: ', `${url}`);
        const result = await axios.delete(url);
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return 'Skill not been deleted';
    }
}

export async function uploadResume(resume) {
    try {
        const verification = await verifyTokenApi();
        if (verification === 'OK') {
            const url = '/resume';
            console.log('[POST] processing url: ', `${url}`);
            return await axios.post(url, resume);
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with this server! Returning default response.');
        return 'Resume not been uploaded';
    }
}


export async function putSkills(skills) {
    try {
        const verification = await verifyTokenApi();
        if (verification === 'OK') {
            const url = `${apiPublic}/api/skills`;
            console.log('[PATCH] processing url: ', `${url}`);
            const result = await axios.put(url, {skills});
            return result.data;
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return 'Skill not been patched';
    }
}


export async function uploadBiography(biography) {
    try {
        const verification = await verifyTokenApi();
        if (verification === 'OK') {
            const url = '/biography';
            console.log('[POST] processing url: ', `${url}`);
            return await axios.post(url, {biography});
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with this server! Returning default response.');
        return 'Resume not been uploaded';
    }
}

export async function getBiography() {
    try {
        const url = `${selfUrl}/biography`;
        console.log('[GET] processing url: ', `${url}`);
        const result = await axios.get(url);
        return result.data.biography;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with this server! Returning default response.');
        return 'Biography not been uploaded';
    }
}


export const getSkillsByPanes = skills => {
    const sortedSkills = skills.slice().sort(sortByOrder);
    return sortedSkills.reduce((acc, curr) => {
        if (curr['panel'] === 'left') {
            acc['left'].push(curr);
        }
        if (curr['panel'] === 'right') {
            acc['right'].push(curr);
        }
        return acc;
    }, {left: [], right: []});
};

export const sortByOrder = (a, b) => {
    if (a.order * 1 < b.order * 1) {
        return -1;
    }
    if (a.order * 1 > b.order * 1) {
        return 1;
    }
    return 0;
};
