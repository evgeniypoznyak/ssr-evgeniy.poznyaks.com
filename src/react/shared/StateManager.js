import {createContext} from 'react';

export const sortSkills = skills => {
    const sortedSkills = skills.slice().sort(compareByOrder);
    return sortedSkills.reduce((acc, curr) => {
        if (curr['panel'] === 'left') {
            acc['left'].push(curr);
        }
        if (curr['panel'] === 'right') {
            acc['right'].push(curr);
        }
        return acc;
    }, {left: [], right: []});
};

export const compareByOrder = (a, b) => {
    if (a.order < b.order) {
        return -1;
    }
    if (a.order > b.order) {
        return 1;
    }
    return 0;
};

export const State = createContext(null);
