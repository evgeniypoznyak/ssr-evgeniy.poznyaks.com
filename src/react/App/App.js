import React, {useContext, useEffect} from 'react';
import reactRoutes from '../reactRoutes';
import {Route, Switch} from 'react-router-dom';
import NoMatch from '../components/404/NoMatch';
import Layout from '../hoc/Layout/Layout';
import {verifyToken} from '../shared/api';
import {State} from '../shared/StateManager';


const App = props => {
    const context = useContext(State);
    useEffect(() => {
        (async () => {
            const res = await verifyToken();
            if (res === 'OK') {
                context.setData({...context.data, 'authorized': true});
            }
        })();
    }, []);

    const routesWithReact = <Switch>
        {reactRoutes.map(({path, exact, component: Component, ...rest}) => (
            <Route key={path} path={path} exact={exact} render={props => (
                <Component {...props} {...rest} />
            )}/>
        ))}
        <Route render={props => <NoMatch {...props} />}/>
    </Switch>;

    return (
        <Layout context={props.context}>
            {routesWithReact}
        </Layout>
    );
};
export default App;
