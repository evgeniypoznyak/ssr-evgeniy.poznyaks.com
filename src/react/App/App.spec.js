import React from 'react';
import {mount, shallow} from 'enzyme';
import App from './App';
import {MemoryRouter, Route} from 'react-router-dom';
import {getSkillsByPanes, skills} from '../shared/api';
import Layout from '../hoc/Layout/Layout';

let wrapper;
let mountWrapper;

const context = {
    data: {
        rawData: skills,
        sortedData: getSkillsByPanes(skills),
    },
};

beforeAll(() => {
    wrapper = shallow(<App/>);
    mountWrapper = mount(<MemoryRouter><App context={context}/></MemoryRouter>);
});

describe('<App/>', () => {
    it('renders correctly', () => {
        expect(wrapper.children.length).toBe(1);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should render App with routes and children', () => {
        const route = mountWrapper.find(Route);
        expect(route).toHaveLength(2);
        expect(route.children).toHaveLength(1);
    });

    it('should render App with Layout and children', () => {
        const layout = mountWrapper.find(Layout);
        expect(layout).toHaveLength(1);
        expect(layout.children).toHaveLength(1);
    });
});


