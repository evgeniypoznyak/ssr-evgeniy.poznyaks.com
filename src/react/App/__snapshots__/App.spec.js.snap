// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly 1`] = `
<Layout>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/skills/:id"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</Layout>
`;
